ui/package.json

```diff
@@ -8,7 +8,7 @@
     "scripts": {
         "build": "rm -Rf dist && NODE_OPTIONS='--openssl-legacy-provider' NODE_ENV=production webpack --mode production --config ./src/app/webpack.config.js",
         "start": "NODE_OPTIONS='--no-experimental-fetch --openssl-legacy-provider' webpack-dev-server --config ./src/app/webpack.config.js",
-        "lint": "eslint --fix ./src/app",
+        "lint": "eslint --fix ./src/app && tsc --noEmit",
```

Comment:
Just to clarify any type checking failures will not be caught until lint is run now?

Reply:
I am wondering if SWC is a better option than esbuild if thats the case?

Reply:
Build won't type-check anymore, correct. This adds it to the `lint` script and the tests do still run with `ts-jest` (but they have very little coverage, and it's ripe to make faster with a different plugin too).
This has basically emerged as the most common pattern for TS codebases in recent years. Build with Babel or `esbuild`, type-check with `tsc`. Since I maintained TSDX I was at the forefront of that a few years ago and still maintain `rollup-plugin-typescript2`, which hasn't had as much usage recently in part due to this pattern (and Vite & `esbuild` popularity in general).

Re: SWC, it also does not support type-checking, although there is an up and coming Rust library for that: https://github.com/swc-project/swc/issues/571.
So basically type-checking is currently limited to the TS compiler itself (and its integrations); all other TS-compatible build tools do not perform type-checking.

I also currently prefer `esbuild` over SWC as it tries to read `tsconfig.json` whereas SWC has its own configuration (it is similar to Babel's config in principle, but not compatible). (`esbuild` is also more popular and has more features currently, afaik)

Reply:
Hmm okay, thanks for the explanation. I personally would like to get type checking but if that is the current pattern, I am happy to go with it. Perhaps  we could add an option to make it type check as well in the future, but happy for that to be a new feature in the future.

Reply:
To be explicit, we do have type-checking, just not during `yarn build` specifically. 

I don't really have a strong opinion on the pattern (I didn't in the above discussions a few years ago either), but the thought process makes sense, where you get your code working first and then you make sure your types, static analysis, tests, etc all pass. Although certainly you can get bugs because those are failing.

For full incremental development, some people also run 4 separate terminals simultaneously: dev build in watch mode, type checker in watch mode, linter in watch mode, and tests in watch mode. Separating out all the concerns explicitly that way.

Also we actually need to improve the strictness of the `tsconfig.json` first, as it's pretty loose. Unfortunately it does require independently building `argo-ui` before we can do that, since the settings have to match right now per https://github.com/argoproj/argo-ui/issues/453. That would also probably result in a large performance improvement for builds similar to this PR.